conans/cli/commands/list.py

```diff
@@ -3,31 +3,40 @@
 from conans.cli.command import conan_command, conan_subcommand, Extender, COMMAND_GROUPS
 from conans.cli.output import cli_out_write
 from conans.client.output import Color
-from conans.model.ref import PackageReference
+from conans.errors import ConanException, InvalidNameException
+from conans.model.ref import PackageReference, ConanFileReference
 from conans.util.dates import iso8601_to_str
 
 remote_color = Color.BRIGHT_BLUE
 recipe_color = Color.BRIGHT_WHITE
 reference_color = Color.WHITE
+error_color = Color.BRIGHT_RED
+field_color = Color.BRIGHT_YELLOW
+value_color = Color.CYAN
 
 
-def _print_common_headers(result, ref_type):
-    if result.get("error"):
-        # TODO: Handle errors
-        return
-
+def _print_common_headers(result):
     if result.get("remote"):
         cli_out_write(f"{result['remote']}:", fg=remote_color)
     else:
         cli_out_write("Local Cache:", remote_color)
 
-    if not result.get("results"):
-        cli_out_write(f"There are no matching {ref_type}", indentation=2)
-
 
 def list_recipes_cli_formatter(results):
+    no_results_msg = "There are no matching recipe references"
     for result in results:
-        _print_common_headers(result, "recipes")
+        _print_common_headers(result)
+        if result.get("error"):
+            error = result['error']
+            if error.startswith("NotFoundError:"):
```

Comment:
This error message pattern seems fragile, seems connected to the below comments about exceptions.